dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/index/IndexServiceBean.java

```diff
@@ -1173,12 +1168,11 @@ private String getTopicClassificationTermOrTermAndVocabulary(DatasetField topicC
         return finalValue;
     }
 
-    public List<String> findPathSegments(Dataverse dataverse, List<String> segments) {
-        Dataverse rootDataverse = findRootDataverse();
+    public List<String> findPathSegments(Dataverse dataverse, List<String> segments, Dataverse rootDataverse) {
         if (!dataverse.equals(rootDataverse)) {
```

Comment:
We don't need to pass extra argument: `rootDataverse`. It would be enough to do `if (!dataverse.isRoot())`